PWG/EMCAL/EMCALbase/AliAnalysisTaskEmcalEmbeddingHelper.cxx

```diff
@@ -615,6 +623,83 @@ bool AliAnalysisTaskEmcalEmbeddingHelper::IsRunInRunlist(const std::string & pat
   return false;
 }
 
+/**
+ * Get the data background run number from the data file path, fRunNumberBgk and 
+ * Determine the MC run block range,  fEmbeddedRunblockMin and  fEmbeddedRunblockMax (used in IsRunInRunblock())
+ * Right now only done once, for the first data file
+ */
+void AliAnalysisTaskEmcalEmbeddingHelper::SetRunblockRange() 
+{
+  Int_t nBlocks = fEmbeddedRunblock.size();
+  
+  if ( nBlocks == 0 ) return ;
+  
+  AliDebug(1,Form("AliAnalysisTaskEmcalEmbeddingHelper::SetRunblockRange() - nRunBlocks %d\n", nBlocks));
+  
+  // Recover the run number from the input data file path.
+  //
+  // InputEvent() seems to be connected to the external event
+  // so one cannot use InputEvent()->GetRunNumber()
+  AliAnalysisManager *mgr = AliAnalysisManager::GetAnalysisManager();
+   
+   if ( !mgr ) return ;
+  
+  Int_t runNumber = mgr->GetRunFromPath(); 
+  if ( fDataRunNumber != runNumber )
+  {
+    //printf("\t Run number, new %d, previous %d\n",runNumber,fDataRunNumber);
+    fDataRunNumber = runNumber;
+  }
```

Comment:
To get the internal event (and the run number), please use AliAnalysisTaskSE::InputEvent()

Reply:
I tried, it does not work, I have the impression it takes the input event number from the external event and at this stage it is not yet set, I get 0, that is why I am using this. Maybe there is something wrong in my setup.

Reply:
If this is called during `UserExec()` (which is should for your case), it has to be defined. Note that it must be `AliAnalysisTaskSE::InputEvent()`, not just `InputEvent()` (the latter **will** get the external event)

Reply:
Checked and it works, I am about to commit